### PR TITLE
Link plate study

### PR DESCRIPTION
**Pull request by josenavas**
Merge #651 first to see my changes.

Some of the assumptions about the data changed from the original DB design. In this PR I'm adding a link between the study and the plate so we know which studies are going to be plated.

### CODE REVIEW

File: amgut/db/patches/0041.sql [hunk 1]
```diff
@@ -116,14 +116,13 @@ CREATE INDEX idx_sample_plate_layout_sample_id ON pm.sample_plate_layout ( sampl
...
 CREATE TABLE pm.study (
-	study_id             bigserial  NOT NULL,
-	qiita_study_id       bigint  ,
-	title                varchar  ,
-	alias                varchar  ,
-	notes                varchar  ,
+	study_id             bigint   NOT NULL,
+	title                varchar  NOT NULL,
+	alias                varchar  NOT NULL,
+    jira_key			 varchar  NOT NULL,
```
Comment (antgonza) [inline amgut/db/patches/0041.sql]: Can you fix the indentation?

Comment (antgonza) [inline amgut/db/patches/0041.sql]: Note that in JIRA you need the internal project id to instantiate and, for example, delete a project, not the key.

File: amgut/db/patches/0041.sql [hunk 2]
```diff
@@ -135,6 +134,15 @@ CREATE TABLE pm.study_sample (
 CREATE INDEX idx_study_sample_study_id ON pm.study_sample ( study_id );
 CREATE INDEX idx_study_sample_sample_id ON pm.study_sample ( sample_id );
 
+CREATE TABLE pm.sample_plate_study (
+	sample_plate_id		bigint NOT NULL,
+	study_id			bigint NOT NULL,
```
Comment (antgonza) [inline amgut/db/patches/0041.sql]: Fix tabs?

File: amgut/db/patches/0041.sql (hunk 1 above)
Comment (josenavas) [inline amgut/db/patches/0041.sql]: weird, on my end was looking fine - fixing.

Comment (josenavas) [inline amgut/db/patches/0041.sql]: thanks!

File: amgut/db/patches/0041.sql (hunk 2 above)
Comment (josenavas) [inline amgut/db/patches/0041.sql]: Done

File: amgut/db/patches/0041.sql [hunk 3]
```diff
@@ -1,201 +1,209 @@
...
-	title                varchar  ,
-	alias                varchar  ,
-	notes                varchar  ,
-	CONSTRAINT pk_study PRIMARY KEY ( study_id ),
-	CONSTRAINT uq_study_title UNIQUE ( title ) ,
-	CONSTRAINT uq_study_qiita_study_id UNIQUE ( qiita_study_id )
+    study_id             bigint   NOT NULL,
+    title                varchar  NOT NULL,
+    alias                varchar  NOT NULL,
+    jira_id              varchar  NOT NULL,
```
Comment (antgonza) [inline amgut/db/patches/0041.sql]: Here is jira_id and ...

File: amgut/db/ag.dbs [hunk 4]
```diff
@@ -22943,20 +22919,35 @@ AS $function$xpath_exists$function$
...
 		<table name="study" >
-			<column name="study_id" type="bigserial" jt="-5" mandatory="y" />
-			<column name="qiita_study_id" type="bigint" jt="-5" />
-			<column name="title" type="varchar" jt="12" />
-			<column name="alias" type="varchar" jt="12" />
-			<column name="notes" type="varchar" jt="12" />
+			<column name="study_id" type="bigint" jt="-5" mandatory="y" />
+			<column name="title" type="varchar" jt="12" mandatory="y" />
+			<column name="alias" type="varchar" jt="12" mandatory="y" />
+			<column name="jira_key" type="varchar" jt="12" mandatory="y" />
```
Comment (antgonza) [inline amgut/db/ag.dbs]: ... here is jira_key ... which one is it?

Comment (josenavas) [inline amgut/db/ag.dbs]: Good catch - I've updated the patch but not the dbschema file! Fixed - everything now should say jira_id.

File: amgut/db/ag.dbs [hunk 5]
```diff
@@ -23167,5 +23156,35 @@ AS $function$xpath_exists$function$
...
 			<entity schema="pm" name="tm50_8_tool" />
 		</group>
+		<script name="Sql" id="SQL4682242" confirmed="y" >
+			<string><![CDATA[ALTER TABLE pm.study DROP COLUMN qiita_study_id;
+
+ALTER TABLE pm.study DROP COLUMN notes;
+
+ALTER TABLE pm.study DROP CONSTRAINT uq_study_qiita_study_id;
+
+ALTER TABLE pm.study ADD jira_key varchar  NOT NULL;
```
Comment (antgonza) [inline amgut/db/ag.dbs]: ?????

Comment (josenavas) [inline amgut/db/ag.dbs]: That's so weird? Where is that coming from?

**Pull request merged**